cli/repl.rs

```diff
@@ -32,6 +38,82 @@ impl Validator for Helper {
   }
 }
 
+impl Highlighter for Helper {
+  fn highlight_hint<'h>(&self, hint: &'h str) -> Cow<'h, str> {
+    hint.into()
+  }
+
+  fn highlight<'l>(&self, line: &'l str, pos: usize) -> Cow<'l, str> {
+    self.highlighter.highlight(line, pos)
+  }
+
+  fn highlight_candidate<'c>(
+    &self,
+    candidate: &'c str,
+    _completion: rustyline::CompletionType,
+  ) -> Cow<'c, str> {
+    self.highlighter.highlight(candidate, 0)
+  }
+
+  fn highlight_char(&self, line: &str, _: usize) -> bool {
+    !line.is_empty()
+  }
+}
+
+struct LineHighlighter {
+  regex: Regex,
+}
+
+impl LineHighlighter {
+  fn new() -> Self {
+    let regex = Regex::new(
+      r#"(?x)
+      (?P<comment>(?:/\*[\s\S]*?\*/|//[^\n]*)) |
+      (?P<string>(?:"([^"\\]|\\.)*"|'([^'\\]|\\.)*'|`([^`\\]|\\.)*`)) |
+      (?P<regexp>/(?:(?:\\/|[^\n/]))*?/[gimsuy]*) |
+      (?P<number>\d+(?:\.\d+)*(?:e[+-]?\d+)*n?) |
```

Comment:
@caspervonb  `(?:\.\d+)*` should be `(?:\.\d+){0, 1}` to prevent numbers from having more that one dot

Reply:
As for your deleted comment.

> (?:\.\d+){1}

Would not capture integers.

> (?:\.\d+){0, 1}

Sure, could also be a ? quantifier.